### ISSUE TITLE
AJAX request not valid

### ISSUE THREAD

**Issue by kuyucakli** [skoczen/django-ajax-uploader#52]
Hi ,

I have installed fineuploader.js and try to upload files but i get error AJAX request not valid?

I think the error is because of GET['qqfile'] ... 
  try:
                    filename = request.GET['qqfile']
                except KeyError:
                    return HttpResponseBadRequest("AJAX request not valid")

How can i get rid of this error?  Thanks

**Comment by mbaechtold** [comment]
Yes, `request.GET` is empty: `<QueryDict: {}>` but `request.POST` contains the data:

`<QueryDict: {u'qqtotalfilesize': [u'10485760'], u'qqfilename': [u'10_mb_file.bin'], u'qquuid': [u'f613fb9b-0389-4029-8ff1-30821b664493']}>`

I'm using fineupload 4.3.1.

I haven't had the time to look for a solution.

**Comment by skoczen** [comment]
I believe this has been fixed with 0.3.4.  We're using it with the latest fineuploader, at least.  Please reopen if you still see this!

Broadly, you could fix the error above by using `request.REQUEST` or `request.POST` instead.

I'm also doing a catch-up clean-out of issues on ajaxuploader. Looks like this one's stalled, so I'm closing it. Please re-open if that's not right!

Thanks,
-Steven

**Comment by mbaechtold** [comment]
Hi @skoczen 

I just upgrade to 0.3.5 and the problem still exists. The problem is in https://github.com/GoodCloud/django-ajax-uploader/blob/master/ajaxuploader/views/base.py#L33

I'm using Fine Uploader 4.3.1 and when I upload a file, `request.is_ajax()` is true but there is no GET parameter called `qqfile`. However, the file is available from `request.POST['qqfilename']`.

**Comment by skoczen** [comment]
Added a fix for this in 0.3.6.  Not sure why you're seeing it and I'm not in a number of sites, but since I've got a minute and your fix looks reasonable, just putting it in!

``` python
if 'qqfile' in request.GET:
    filename = request.GET['qqfile']
else:
    filename = request.REQUEST['qqfilename']
```
